### ISSUE TITLE
User Interface Timezone Setting Not Saving

### ISSUE THREAD

**Issue by teschste-reyrey** [dbeaver/dbeaver#22947]
### Description

In Windows -> Preferences -> User Interface, when I select a specific time zone from the dropdown (America/Chicago), apply it and close settings, then re-check the setting immediately after, the setting has returned to "Default".

I tried restarting DBeaver after selecting/applying a specific time zone as well, since it indicates that DBeaver must be restarted to take effect, but that did not save the setting either.

Am I missing something?  This is the only place I see to set the desired time zone.

Thanks!

### DBeaver Version

Community Edition 23.3.5.202402181738

### Operating System

Windows 10 64-bit

### Database and driver

PostgreSQL 13.10 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44), 64-bit
PostgreSQL JDBC Driver

### Steps to reproduce

1. Go to Windows -> Preferences -> User Interface.
2. Select time zone America/Chicago (will likely fail with any time zone).
3. Click Apply, then Apply and Close (it may not be necessary to select both but I prefer to be certain).
4. Close/reopen DBeaver.
5. Go to Windows -> Preferences -> User Interface and see that the time zone selection has returned to "Default" (or possibly whatever it was set to previously).

**Comment by Destrolaric** [comment]
@teschste-reyrey, Hi, is DBeaver installed via the Windows Store?

**Comment by teschste-reyrey** [comment]
No, it was originally installed from, and received updates from, https://dbeaver.io/download/.  Recently we had an internal issue where downloads from that site were blocked (now fixed) and I downloaded the current version from this site, then last week the application auto-updated from the original site again.

**Comment by Destrolaric** [comment]
@teschste-reyrey 
Hi, could you see if the changed timezone was saved in: 
```
C:\Users\your_username\AppData\Roaming\DBeaverData\workspace6\.metadata\.plugins\org.eclipse.core.runtime\.settings\org.jkiss.dbeaver.core.prefs
```

**Comment by teschste-reyrey** [comment]
OK, that is weird.  I checked that file and found no reference to timezone.  I tried setting the value again and the file was touched (timestamp updated) but there was still no reference to timezone.  I tried it again using America/New_York and this time it did save in that file as ```java.client.timezone=America/New_York```.  I then tried America/Chicago again and it saved as well.

Curious, I changed it back to Default but now it seems stuck on America/Chicago.

Next, with DBeaver closed, I removed the timezone line from the file, and when I launched DBeaver I was back to Default.  I then tried several times, setting it to America/Chicago and it never saved to the file.  Then I changed it to America/New_York again and it did save.  After that, I was again able to set it to America/Chicago and it would save.

It seems that at least America/Chicago will not save to the file if the ```java.client.timezone``` line does not already exist, and if the line does exist, setting the value back to Default does not remove the line from the file.

**Comment by serjiokov** [comment]
Hi @teschste-reyrey 
So, let me explain the concept of preferences. 
First of all the file of preferences related to the workspace (by defult name of workspace is ".../workspace6"), you might have different sets of settings depending on your projects and needs. The next preference asset location depends on access right on your PC, it might be stored locally or globally.
I tested on the latest build and the specification of the time zone is successful for me, all works as expected.
Could you please, try the latest build and let me know your result?

**Comment by teschste-reyrey** [comment]
Even on the latest version, it still works as I described in my comment on March 7.

**Comment by brane53** [comment]
I'm experiencing this same issue as @teschste-reyrey described. I'm running the latest version of DBeaver on mac

**Comment by serjiokov** [comment]
Tested save time zone to the preferences file. No issue.
Please keep in mind you should not prevent writing operations to the file (require to have w+r permissions), 
the product restart requires in order to apply these changes.

**Comment by E1izabeth** [comment]
It's been a while since no update here. If the issue is still actual please let me know and provide additional information.

**Comment by teschste-reyrey** [comment]
The issue still exists as described above.  In short, if there is no "java.client.timezone" entry in file `C:\Users\teschste\AppData\Roaming\DBeaverData\workspace6\.metadata\.plugins\org.eclipse.core.runtime\.settings\org.jkiss.dbeaver.core.prefs`, when I attempt to set the tine zone to America/Chicago and apply the change, it will not save, in other words it will not create the java.client.timezone entry in the prefs file.  However, if I select a different time zone, for example America/Newr_York, and apply it, it does create the java.client.timezone entry in the prefs file, at which point I can then go back and change the time zone to America/Chicago and it will save correctly.

I do not know if this is an issue specific to the America/Chicago time zone or if it is specific to the time zone of the user (I am in America/Chicago) but this is still a valid issue that should be addressed.

**Comment by Zaccarihas** [comment]
Have the same issue when trying to change to timezone 'Europe/Stockholm' on both my windows 11 computer and on my Linux Mint machine. Applying  the timezone in the Preferences/User Interface and then restart dbeaver will only return to Default setting.

**Comment by bankmonitor-gabor-viczai** [comment]
Same here. Linux, Ububtu 24.04, unable to set time zone, it always sets back to Default.

**Comment by arhayka** [comment]
We can't reproduce this issue.
@Zaccarihas @bankmonitor-gabor-viczai, which version are you using?
Could you please try reproducing this on the latest DBeaver version, 24.3.1?

**Comment by Zaccarihas** [comment]
Hi,

I was using an old version but I've now installed version 24.3.1, and the
problems remain. I've attached a clip that show the failing procedure.

BR
Anders
 movie.mp4.zip
<https://drive.google.com/file/d/1Hc9WUr-pBfhzWuZyZmNjZEk1Oo4ylEzI/view?usp=drive_web>

**Comment by E1izabeth** [comment]
Unfortunately, GitHub can't attach files when you answer by email, so I don't see the files you sent. Can you please go to GitHub and attach them there?

**Comment by arhayka** [comment]
This is a video from @Zaccarihas 
https://github.com/user-attachments/assets/b829e101-6357-4f3a-989e-4a13b28f4ae4

**Comment by dutompson** [comment]
> This is a video from @Zaccarihas https://github.com/user-attachments/assets/b829e101-6357-4f3a-989e-4a13b28f4ae4

I just found out that if you type in the search box the timezone name, it won't persist the chosen value, but if you look for it in the long list and select it without typing in, it will save it properly.

But even after this, when I query the db, I'm getting the default timezone still.

**Comment by Matvey16** [comment]
Have you tried resetting the workspace or changing the workspace location?

**Comment by E1izabeth** [comment]
It's been a while since there has been an update here. Please let me know if the issue is still present and answer the question.

**Comment by teschste** [comment]
I renamed my workspace folder and re-launched DBeaver.  A new workspace was created and I re-tested exactly as I originally described above, and the America/Chicago time zone is not saved.

**Comment by teschste** [comment]
I see you just closed this issue.  So you are not going resolve the issue, even though myself and others can duplicate it at will?

**Comment by E1izabeth** [comment]
Hi, @teschste
Thanks for the update.
We usually close stale issues after a month of waiting for an answer and reopen them as soon as the answer appears.
I hope we'll figure out the roots of such behavior. Right now, we can't reproduce it. However, logs might help. So, please send us the DBeaver log files. The log files are generated automatically and called .log and .dbeaver-debug.log exactly. They are located in the %APPDATA%\DBeaverData\workspace6\\.metadata directory. You can collect them to archive automatically with `Help->Collect Diagnostic Info`
![image](https://github.com/user-attachments/assets/51c7ca2e-6be2-489e-b495-028b9aa0e386)

Here is our wiki page on this matter: [dbeaver.com/docs/wiki/Log-files/#log-files](http://dbeaver.com/docs/wiki/Log-files/#log-files)

**Comment by teschste** [comment]
I used the new workspace that DBeaver created as noted above.  Diagnostic logs attached.

[dbeaver-diagnostic-info-1738866515596.zip](https://github.com/user-attachments/files/18694599/dbeaver-diagnostic-info-1738866515596.zip)

**Comment by Destrolaric** [comment]
`2025-02-06 12:28:03.013 - Platform not initialized: metadata folder may be not set`
Might be the part of the problem.

**Comment by arhayka** [comment]
We decided to add additional logs for timezone setting to understand what happens.

**Comment by serge-rider** [comment]
A couple of config save/read fixes applied.
Extra logging added.

If this issue will remain please check debug log for messages with `time zone` mentions.

**Comment by teschste** [comment]
I just updated to the latest version 24.3.5.202502162012, and the issue is now resolved.  Thank you!
